lib/spack/spack/paths.py

```diff
@@ -20,59 +21,59 @@
 spack_root = prefix
 
 #: bin directory in the spack prefix
-bin_path = os.path.join(prefix, "bin")
+bin_path = str(PurePath(prefix, "bin"))
 
 #: The spack script itself
-spack_script = os.path.join(bin_path, "spack")
+spack_script = str(PurePath(bin_path, "spack"))
 
 #: The sbang script in the spack installation
-sbang_script = os.path.join(bin_path, "sbang")
+sbang_script = str(PurePath(bin_path, "sbang"))
 
 # spack directory hierarchy
-lib_path = os.path.join(prefix, "lib", "spack")
-external_path = os.path.join(lib_path, "external")
-build_env_path = os.path.join(lib_path, "env")
-module_path = os.path.join(lib_path, "spack")
-command_path = os.path.join(module_path, "cmd")
-analyzers_path = os.path.join(module_path, "analyzers")
-platform_path = os.path.join(module_path, "platforms")
-compilers_path = os.path.join(module_path, "compilers")
-build_systems_path = os.path.join(module_path, "build_systems")
-operating_system_path = os.path.join(module_path, "operating_systems")
-test_path = os.path.join(module_path, "test")
-hooks_path = os.path.join(module_path, "hooks")
-opt_path = os.path.join(prefix, "opt")
-share_path = os.path.join(prefix, "share", "spack")
-etc_path = os.path.join(prefix, "etc", "spack")
+lib_path = str(PurePath(prefix, "lib", "spack"))
+external_path = str(PurePath(lib_path, "external"))
+build_env_path = str(PurePath(lib_path, "env"))
```

Comment:
What's the difference between `str(PurePath())` and `os.path.join`? Is it because `prefix` is POSIX-style on Windows? Could we avoid changing all these lines by possibly just editing `prefix =...` above (at which point `os.path.join` would produce a consistent result in all these cases)?

Reply:
`str(PurePath())` converts the posix-style path to a windows path. I can change it back to using `os.path.join` for everything after prefix since it'll have the same behavior as all these casts.